### DISCUSSION TITLE
Automatically Calculate Date/Time

### DISCUSSION THREAD

**Discussion by beaugast** [Budibase/budibase#4344] [Help]
I have a Start Date/Time field and a Stop Date/Time field and I'm trying to get the Stop Date/Time field to auto-populate the same date as the Start Date and use the start time to add 2 hours to the Stop Time. What's the best way to do this? I've tried creating a binding with no luck. Probably something I'm just completely missing.

**Comment by mjashanks** [comment]
Hi There.

On your "Stop" datepicker, set the `Default Value` binding to the following Javascript

```javascript
if (!$("New Form.Fields.start")) return;

const two_hours_in_milliseconds = 2*60*60*1000;

const start = new Date($("New Form.Fields.start"));

return start.setTime(start.getTime() + two_hours_in_milliseconds);
```

you'll need to swap out `$("New Form.Fields.start")` for whatever your `Start` fieldname is.

**Reply by GlitchWitch** [reply]
This worked for me! Thanks @mjashanks 

For anyone looking at this in the future, you can use the following slightly different formula to add days instead of hours. 

```js
if (!$("Form.Fields.Start Date")) return;
const fourteen_days_in_milliseconds = 14*24*60*60*1000;
const start = new Date($("Form.Fields.Start Date"));
return start.setTime(start.getTime() + fourteen_days_in_milliseconds);
```
